src/nvim/os/lang.c

```diff
@@ -17,14 +17,32 @@ void lang_init(void)
 {
 #ifdef __APPLE__
   if (os_getenv("LANG") == NULL) {
+    const char *lang_region = NULL;
+    CFTypeRef cf_lang_region = NULL;
+
     CFLocaleRef cf_locale = CFLocaleCopyCurrent();
-    CFTypeRef cf_lang_region = CFLocaleGetValue(cf_locale,
-                                                kCFLocaleIdentifier);
-    CFRetain(cf_lang_region);
-    CFRelease(cf_locale);
+    if (cf_locale) {
+      cf_lang_region = CFLocaleGetValue(cf_locale, kCFLocaleIdentifier);
+      CFRetain(cf_lang_region);
+      lang_region = CFStringGetCStringPtr(cf_lang_region,
+                                          kCFStringEncodingUTF8);
+      CFRelease(cf_locale);
+    } else {
+      // Use the primary language defined in Preferences -> Language & Region
+      CFArrayRef cf_langs = CFLocaleCopyPreferredLanguages();
+      if (cf_langs && CFArrayGetCount(cf_langs) > 0) {
+        cf_lang_region = CFArrayGetValueAtIndex(cf_langs, 0);
+        CFRetain(cf_lang_region);
+        CFRelease(cf_langs);
+        lang_region = CFStringGetCStringPtr(cf_lang_region,
+                                            kCFStringEncodingUTF8);
+      } else {
+        ELOG("$LANG is empty and your primary language cannot be inferred. "
+             "Falling back to \"C\".");
+        lang_region = "C";
```

Comment:
@bfredl suggested "return and do nothing" here. Is that preferable?

Reply:
My point was that on other Unix we don't do anything in this case in this function. By "return and do nothing" we simply follow the same code path as other Unix to handle this, which already works well enough.